### PR TITLE
clipboard feature for model matrices

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
Improving both parsing and matching semantics, we can now take
- TSV or CSV
- Raw data, or data with column/row labels

Note: I was thinking of doing this in `utils/text.ts` but changed my mind, but I tightened up the typings there so going to squeeze this in.

### Testing
Goto `http://localhost:8080/clipboard` and try to paste in different matrices, e.g.

```
,I_unvax_0_9,I_unvax_10_19
S_unvax_0_9,33,44
S_unvax_10_19,55,66
```

or 

```
11,22,11
88,99,88
```

Also `yarn workspace hmi-client run test dsv`

### CODE REVIEW

File: packages/client/hmi-client/src/components/dataset/tera-dataset-description.vue [hunk 1]
```diff
@@ -47,7 +47,7 @@
 				<div v-for="(value, key) in dataset.metadata" :key="key" class="row">
 					<template v-if="key.toString() !== 'preview'">
 						<div class="col key">
-							{{ snakeToCapitalized(key) }}
+							{{ snakeToCapitalized(key.toString()) }}
```
Comment (Tom-Szendrey) [inline packages/client/hmi-client/src/components/dataset/tera-dataset-description.vue:50]: Seems unrelated, just wanted to make sure this is intentional

**Comment by Tom-Szendrey** [comment]
(Reasonably) this fails when pasting really ugly non uniform data:
```
1,1
2,2,2,2
3,3
4
```

**Review by Tom-Szendrey — APPROVED** [review]
Sweet!

File: packages/client/hmi-client/src/components/dataset/tera-dataset-description.vue (hunk 1 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/components/dataset/tera-dataset-description.vue:50]: Not intentional - I started out working with `utils/text.ts` which is where the function came from, but changed my mind and moved to a separate dsv-util, I thought I would just leave existing changes in since they are just type-cleanups.

File: packages/client/hmi-client/tests/unit/utils/dsv.spec.ts
Comment (shawnyama) [inline packages/client/hmi-client/tests/unit/utils/dsv.spec.ts:1]: file name misspell - dsv -> csv

Comment (shawnyama) [reply packages/client/hmi-client/tests/unit/utils/dsv.spec.ts:1]: nvmind this looks intentional

Comment (shawnyama) [reply packages/client/hmi-client/tests/unit/utils/dsv.spec.ts:1]: delimiter got it

File: packages/client/hmi-client/src/temp/Clipboard.vue [hunk 2]
```diff
@@ -31,71 +32,46 @@ const mmt = ref<MiraModel>(emptyMiraModel());
...
+	const parseResult = dsvParse(text);
+	const matrices = createParameterMatrix(mmt.value, mmtParams.value, 'beta');
+	const matrix = matrices[matrixType];
+
+	console.log(parseResult.entries);
+
+	matrix.matrix.forEach((row) => {
+		row.forEach((matrixEntry) => {
+			// If we have label information, use them as they may be more accurate, otherwise use indices
+			if (parseResult.hasColLabels && parseResult.hasColLabels) {
```
Comment (shawnyama) [inline packages/client/hmi-client/src/temp/Clipboard.vue:49]: this checks for the same thing twice

File: packages/client/hmi-client/src/utils/dsv.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			hasRowLabels = true;
+		} else {
+			hasColLabels = false;
+			hasRowLabels = false;
+		}
+	} else if (lines.length > 1 && lineOne.length > 1) {
+		if (Number.isNaN(+lineOne[1])) {
+			hasColLabels = true;
+		}
+		const line2 = lines[1];
```
Comment (shawnyama) [inline packages/client/hmi-client/src/utils/dsv.ts:66]: `lineOne`, `line2` - choose a consistently named suffix

**Review by shawnyama — APPROVED** [review]

**Review by asylves1 — APPROVED** [review]

File: packages/client/hmi-client/src/utils/dsv.ts (hunk 3 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/utils/dsv.ts]: done
